### ISSUE TITLE
Create a basic version of the `Scalar` object

### ISSUE THREAD

**Issue by fmmanicacci** [fmmanicacci/nanograd#1]
The `Scalar` object will represent every node of the computational graph and will be in charge of computing the gradient for each of them. The objectif is to define the backbone of this object.

To correctly list all supported operations separately, an enumeration called `Operation` must be implemented. Each operation will have an associated string representing the operation.

## Acceptance criteria

The `Scalar` object must contain the following attributes:

- [x]  `data`, the scalar value
- [x] `label`, the optional label associated to the node
- [x] `requires_grad`, a boolean set to `False` by default that indicates whether the gradient should be computed for the current node
- [x] `_grad`, the gradient of the node, $0$ by default
- [x] `_prev`, the set of nodes that contribute to create the current node
- [x] `_op`, the operator used to combine the node present in `_prev`
- [x] `_backward_fn`, a function reference used to compute the gradient of the `_prev` node during the backward phase

The `Scalar` object must contain the following methods:

- [x] `__str__`, that provide a readable string representation of the object which contains the following information:
   - [x] the label, if it not `None`
   - [x] the data, as a floating point number with 6 decimals
   - [x] the `requires_grad` boolean
   - [x] the gradient, if the attributes `requires_grad` is `True`
   - [x] the operation, as a string if the operation is different than `Operation.NONE`
   - [x] the number of children, if the operation is different than `Operation.NONE`
- [x] `__repr__`, that provide a more information-rich string representation of the object that display the following information, one per line :
   - [x] the label, if it not `None`
   - [x] the data, as a floating point number with 6 decimals
   - [x] the `requires_grad` boolean
   - [x] the gradient, if the attributes `requires_grad` is `True`
   - [x] the operation, as a string if the operation is different than `Operation.NONE`
   - [x] the children, by using the string representation returned by the __str__ method if the number of children is not $0$, else display `"None"`

The `Operation` enumeration must contain the following operations:

- [x] `NONE`, associated to the string `none`
- [x] `IDENTITY`, associated to the string `identity`

## Technical requirements

- [x] Typing
- [x] Comments
- [x] Docstrings
- [x] Unit tests
